ICSharpCode.Decompiler/IL/ControlFlow/LoopDetection.cs

```diff
@@ -385,6 +393,22 @@ bool IsValid(ControlFlowNode node)
 			}
 		}
 
+		/// <summary>
+		/// Extension of ControlFlowGraph.HasReachableExit
+		/// Uses loopContext.GetBreakTargets().Any() when analyzing switches to avoid 
+		/// classifying continue blocks as reachable exits.
+		/// </summary>
+		bool HasReachableExit(ControlFlowNode node) => isSwitch
+			? loopContext.GetBreakTargets(node).Any()
+			: context.ControlFlowGraph.HasReachableExit(node);
+		
+		/// <summary>
+		/// Returns the children in a loop dominator tree, with an optional exit point
+		/// Avoids returning continue statements when analysing switches (because increment blocks can be dominated)
```

Comment:
This function isn't returning statements, but control flow nodes. I guess `Avoids returning continue statements` should be `Avoids returning the target block of continue statements`?
After all, `continue;` is fine within a switch, we just don't want to move the increment block itself.

Reply:
Should be changed to `Avoids returning continue statement target nodes (increment blocks)`